Frameworks/CoreGraphics/CGColorSpace.mm

```diff
@@ -38,135 +39,148 @@
 const CFStringRef kCGColorSpaceITUR_2020 = static_cast<CFStringRef>(@"kCGColorSpaceITUR_2020");
 const CFStringRef kCGColorSpaceROMMRGB = static_cast<CFStringRef>(@"kCGColorSpaceROMMRGB");
 
-static IWLazyClassLookup _LazyUIColor2("UIColor");
+static woc::StrongCF<CGColorSpaceRef> s_rgbColorSpace;
+static woc::StrongCF<CGColorSpaceRef> s_grayColorSpace;
 
-@interface CGNSColorSpace : _CGLifetimeBridgingType
-@end
+struct __CGColorSpace : CoreFoundation::CppBase<__CGColorSpace> {
+    __CGColorSpace(CGColorSpaceModel model, CGColorSpaceRef baseSpace = nullptr, size_t lastIndex = 0)
+        : _colorSpaceModel(model), _baseColorSpace(baseSpace), _lastColorIndex(lastIndex) {
+    }
 
-@implementation CGNSColorSpace
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wobjc-missing-super-calls"
-- (void)dealloc {
-    delete (__CGColorSpace*)self;
-}
-#pragma clang diagnostic pop
-@end
+    __CGColorSpace(CGColorSpaceModel model, CGColorSpaceRef baseSpace, size_t lastIndex, const unsigned char* colorTable)
+        : __CGColorSpace(model, baseSpace, lastIndex) {
+        size_t size = CGColorSpaceGetNumberOfComponents(baseSpace) * (lastIndex + 1);
+        _palette.reset(new char[size]);
+        memcpy(_palette.get(), colorTable, size);
+    }
 
-__CGColorSpace::__CGColorSpace(CGColorSpaceModel model) {
-    object_setClass((id)this, [CGNSColorSpace class]);
+    inline CGColorSpaceModel ColorSpaceModel() const {
+        return _colorSpaceModel;
+    }
 
-    colorSpaceModel = model;
-    palette = NULL;
-    lastColor = 0;
-}
+    inline CGColorSpaceRef BaseColorSpace() const {
+        return _baseColorSpace;
+    }
 
-__CGColorSpace::~__CGColorSpace() {
-    if (palette) {
-        IwFree(palette);
-        palette = NULL;
+    inline size_t LastColorIndex() const {
+        return _lastColorIndex;
     }
-}
+
+private:
+    CGColorSpaceModel _colorSpaceModel;
+    woc::StrongCF<CGColorSpaceRef> _baseColorSpace;
+    std::unique_ptr<char[]> _palette;
+    size_t _lastColorIndex;
+};
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-CGColorSpaceRef CGColorSpaceCreateCalibratedRGB(const CGFloat whitePoint[3],
-                                                const CGFloat blackPoint[3],
-                                                const CGFloat gamma[3],
-                                                const CGFloat matrix[9]) {
-    UNIMPLEMENTED();
-    TraceWarning(TAG, L"CGColorSpaceCreateCalibratedRGB not supported");
-    return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelRGB);
+CFTypeID CGColorSpaceGetTypeID() {
+    return __CGColorSpace::GetTypeID();
 }
 
 /**
  @Status Interoperable
 */
-CGColorSpaceRef CGColorSpaceCreateIndexed(CGColorSpaceRef baseSpace, size_t lastIndex, const unsigned char* colorTable) {
-    __CGColorSpace* ret = new __CGColorSpace(kCGColorSpaceModelIndexed);
+CGColorSpaceRef CGColorSpaceCreateDeviceRGB() {
+    if (!s_rgbColorSpace) {
```

Comment:
just move the const static colorspace in here, so we can use "magic static" to initialize on first use w/o having to check every time